### PR TITLE
Add meta DataContainer to GenericJob

### PR DESCRIPTION
**Pull request by pmrv**
I sometimes want to tag jobs with certain extra information, e.g. want kind of atomic structure is used, where it is derived from etc. Right now I usually encode this in subprojects or the job name, but it can get convoluted and error-prone to retrieve it again (due to converting to and from strings), so I'd like a cleaner solution.  

In this PR I've just added an extra DataContainer to GenericJob that saves anything a user might put, but otherwise does not touch it.  There's a bit of danger that people start putting information there that is currently not tracked in pyiron, but really should be, e.g. something like [this](https://github.com/pyiron/pyiron_atomistics/pull/547).  I personally feel it's still a nice thing to have, but we should probably discuss this in the next pyiron meeting.  Any thoughts already?

I'll add tests and such after our discussions.

### CODE REVIEW

File: pyiron_base/job/generic.py [hunk 1]
```diff
@@ -1232,6 +1247,10 @@ def from_hdf(self, hdf=None, group_name=None):
         if "import_directory" in self._hdf5.list_nodes():
             self._import_directory = self._hdf5["import_directory"]
         self._server.from_hdf(self._hdf5)
+        try:
+            self._meta.from_hdf(self._hdf5)
+        except ValueError:
+            pass # no meta data written for this job previously
```
Comment (jan-janssen) [inline pyiron_base/job/generic.py:1253]: Can this be more elegant by checking if the corresponding group exists in the HDF5 file

**Comment by pmrv** [comment]
Rename `job.meta` to `job.data` or `project.data` to `project.meta`.  Conflict with `job['user/foo']`?

**Comment by liamhuber** [comment]
@pmrv after recent meetings, shall we wait until `GenericJob` looks more like `TemplateJob`, then simply add a `self.storage._meta = DataContainer(table_name='meta')` and accompanying read-only-access property? I like this idea, but I think that's an easier time to make the change. We could even directly do it over in #758 as I feel it's very on-topic for that PR.

**Comment by pmrv** [comment]
> @pmrv after recent meetings, shall we wait until `GenericJob` looks more like `TemplateJob`, then simply add a `self.storage._meta = DataContainer(table_name='meta')` and accompanying read-only-access property? I like this idea, but I think that's an easier time to make the change. We could even directly do it over in #758 as I feel it's very on-topic for that PR.

Yup, that sounds like a good idea.  I'm not urgently depending on this change anyway.

**Pull request closed without merging**